modules/quality_control/jsx/qualityControlIndex.js

```diff
@@ -0,0 +1,238 @@
+import React, {Component} from 'react';
+import Loader from 'Loader';
+import {Tabs, TabPane} from 'Tabs';
+import FilterableDataTable from 'FilterableDataTable';
+import PropTypes from 'prop-types';
+
+class QualityControlIndex extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      ImgData: {},
+      BehavioralData: {},
+      isLoadedImg: false,
+      isLoadedBehavioral: false,
+      imgFilter: {},
+      behavioralFilter: {},
+    };
+    this.fetchData = this.fetchData.bind(this);
+  }
+
+  /**
+   * Modify behaviour of specified column cells in the Data Table component
+   *
+   * @param {string} column - column name
+   * @param {string} cell - cell content
+   * @param {object} row - row content indexed by column
+   *
+   * @return {*} a formated table cell for a given column
+   */
+
+  formatColumn(column, cell, row) {
+      let result = <td>{cell}</td>;
+      switch (column) {
+          case 'MRI Parameter Form':
+              let mpfURL = loris.BaseURL + '/mri_parameter_form/?commentID=' +
+                  row.CommentID + '&sessionID=' + row['Session ID'] +
+                  '&candID=' + row.DCCID;
+              result = <td> <a href={mpfURL}>{cell}</a> </td>;
+          case 'Scan Location':
+            if (cell == 'In Imaging Browser') {
+              let imgURL = loris.BaseURL + '/imaging_browser/viewSession/?sessionID=' +
+                  row['Session ID'];
+              result = <td><a href={imgURL}>{cell}</a></td>;
+            }
+          case 'Tarchive':
+            if (cell == 'In DICOM') {
+                let tarchiveURL = loris.BaseURL +
+                    '/dicom_archive/viewDetails/?tarchiveID=' + row.TarchiveID;
+                return <td><a href = {tarchiveURL}>{cell}</a></td>;
+            } else {
+              result = <td>Missing</td>;
+            }
+      }
+      return result;
+  }
+
+  componentDidMount() {
+    this.fetchData('imaging')
+        .then(() => this.setState({isLoadedImg: true}));
+    this.fetchData('behavioral')
+        .then(() => this.setState({isLoadedBehavioral: true}));
+  }
+
+  fetchData(flag) {
+    if (flag === 'imaging') {
+      return fetch(this.props.ImgDataURL, {credentials: 'same-origin'})
+       .then((resp) => resp.json())
+       .then((ImgData) => this.setState({ImgData}))
+       .catch((error) => {
+         this.setState({error: true});
+      });
+    } else if (flag === 'behavioral') {
+      return fetch(this.props.BehavioralDataURL, {credentials: 'same-origin'})
+       .then((resp) => resp.json())
+       .then((BehavioralData) => this.setState({BehavioralData}))
+       .catch((error) => {
+         this.setState({error: true});
+       });
+    }
+  }
```

Comment:
This function could be condensed to have a single `fetch` request returned. Instead of a 'flag', the requested state to be set could be passed as an argument. The url could also be passed to the `fetchData` function. This would remove a lot of the redundant code.
E.G.
```
       return fetch(url, {credentials: 'same-origin'})
         .then((resp) => resp.json())
         .then((data) => this.setState({[state]: data}))
         .catch((error) => {
           this.setState({error: true});
```

Reply:
@HenriRabalais requested changes were made!